[AUTOCUT] Gradle Check Flaky Test Report for SimpleSearchIT

## Flaky Test Report for `SimpleSearchIT`

Noticed the `SimpleSearchIT` has some flaky, failing tests that failed during **post-merge actions** and **timer trigerred run on main branch**.

### Details

| Git Reference | Merged Pull Request | Build Details | Test Name |
|---------------|----------------------|---------------|-----------|
| main | [null](https://github.com/opensearch-project/OpenSearch/pull/null) | [75132](https://build.ci.opensearch.org/job/gradle-check/75132/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testIndexOnlyFloatField {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| 069c2d805d5e8ec8740c05eb7b5fa44fdfd4ca7a | [18097](https://github.com/opensearch-project/OpenSearch/pull/18097) | [57263](https://build.ci.opensearch.org/job/gradle-check/57263/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| e7ed33f60d275c3482ee7b82007ff8bdb00dbe90 | [17978](https://github.com/opensearch-project/OpenSearch/pull/17978) | [57115](https://build.ci.opensearch.org/job/gradle-check/57115/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| 59a6efdb8055f3ec170834f9905bce9f31ac9529 | [18036](https://github.com/opensearch-project/OpenSearch/pull/18036) | [57036](https://build.ci.opensearch.org/job/gradle-check/57036/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| 8cc58b5e56c81dcdb36d8b0afe8281f69c5fd4d3 | [18058](https://github.com/opensearch-project/OpenSearch/pull/18058) | [57023](https://build.ci.opensearch.org/job/gradle-check/57023/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| eba12faff9a84cc15b083c77e9ecda5f939787f6 | [17370](https://github.com/opensearch-project/OpenSearch/pull/17370) | [56995](https://build.ci.opensearch.org/job/gradle-check/56995/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}` |
| 436038d2be54df8979466841fccc6db4971bcb75 | [18041](https://github.com/opensearch-project/OpenSearch/pull/18041) | [56989](https://build.ci.opensearch.org/job/gradle-check/56989/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| 15b702d93273f333039f8c4b346c772e63ae04f1 | [18042](https://github.com/opensearch-project/OpenSearch/pull/18042) | [56934](https://build.ci.opensearch.org/job/gradle-check/56934/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| caf5d71ff02c5250a820f3ec1b346eff37b1d230 | [18039](https://github.com/opensearch-project/OpenSearch/pull/18039) | [56924](https://build.ci.opensearch.org/job/gradle-check/56924/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| 0d1bb9b1f1a86815f19cb867b0ddab8e1e9d31df | [18035](https://github.com/opensearch-project/OpenSearch/pull/18035) | [56909](https://build.ci.opensearch.org/job/gradle-check/56909/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| e61b6ab06d4ee3b929ffc25ef428e26469c9fab7 | [18005](https://github.com/opensearch-project/OpenSearch/pull/18005) | [56703](https://build.ci.opensearch.org/job/gradle-check/56703/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| d18982cad34bd4482d814287d112185fcaea3004 | [17912](https://github.com/opensearch-project/OpenSearch/pull/17912) | [56693](https://build.ci.opensearch.org/job/gradle-check/56693/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| 3a7d8fa465641bb1f5f5c58075ffbc9fefc8aceb | [17976](https://github.com/opensearch-project/OpenSearch/pull/17976) | [56634](https://build.ci.opensearch.org/job/gradle-check/56634/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| c44d230bba25ce20bad96fd91e3ad241852bd01c | [17749](https://github.com/opensearch-project/OpenSearch/pull/17749) | [56412](https://build.ci.opensearch.org/job/gradle-check/56412/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| 693c7884e2fae72f29b3d484f001f8e210195357 | [17921](https://github.com/opensearch-project/OpenSearch/pull/17921) | [56337](https://build.ci.opensearch.org/job/gradle-check/56337/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| b842e48562d3ee1678dff62ad75a57924fd4a9a8 | [17902](https://github.com/opensearch-project/OpenSearch/pull/17902) | [56242](https://build.ci.opensearch.org/job/gradle-check/56242/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| d29e95c0dbaf5716d128e0177e8151bba7dc959e | [17882](https://github.com/opensearch-project/OpenSearch/pull/17882) | [56063](https://build.ci.opensearch.org/job/gradle-check/56063/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}` |
| 18a3b75fb14217d39700fba367617d37f723293d | [17796](https://github.com/opensearch-project/OpenSearch/pull/17796) | [56046](https://build.ci.opensearch.org/job/gradle-check/56046/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| 967eee15eeb4219f2e7e5275c132cf86deff226a | [17216](https://github.com/opensearch-project/OpenSearch/pull/17216) | [56041](https://build.ci.opensearch.org/job/gradle-check/56041/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"false"}}`<br><br>`org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}` |
| d37cc9b32600faef17b3c898f8c86b0fd7640d72 | [16831](https://github.com/opensearch-project/OpenSearch/pull/16831) | [51449](https://build.ci.opensearch.org/job/gradle-check/51449/testReport/) | `org.opensearch.search.simple.SimpleSearchIT.classMethod` |

The other pull requests, besides those involved in post-merge actions, that contain failing tests with the `SimpleSearchIT` class are:

- [17939](https://github.com/opensearch-project/OpenSearch/pull/17939)
- [17782](https://github.com/opensearch-project/OpenSearch/pull/17782)
- [17896](https://github.com/opensearch-project/OpenSearch/pull/17896)
- [17950](https://github.com/opensearch-project/OpenSearch/pull/17950)
- [17855](https://github.com/opensearch-project/OpenSearch/pull/17855)
- [17868](https://github.com/opensearch-project/OpenSearch/pull/17868)
- [18006](https://github.com/opensearch-project/OpenSearch/pull/18006)
- [17718](https://github.com/opensearch-project/OpenSearch/pull/17718)
- [17769](https://github.com/opensearch-project/OpenSearch/pull/17769)
- [17843](https://github.com/opensearch-project/OpenSearch/pull/17843)
- [17888](https://github.com/opensearch-project/OpenSearch/pull/17888)
- [17984](https://github.com/opensearch-project/OpenSearch/pull/17984)
- [18054](https://github.com/opensearch-project/OpenSearch/pull/18054)
- [18243](https://github.com/opensearch-project/OpenSearch/pull/18243)
- [17447](https://github.com/opensearch-project/OpenSearch/pull/17447)
- [17532](https://github.com/opensearch-project/OpenSearch/pull/17532)
- [17880](https://github.com/opensearch-project/OpenSearch/pull/17880)
- [17893](https://github.com/opensearch-project/OpenSearch/pull/17893)
- [17336](https://github.com/opensearch-project/OpenSearch/pull/17336)
- [17682](https://github.com/opensearch-project/OpenSearch/pull/17682)
- [17711](https://github.com/opensearch-project/OpenSearch/pull/17711)
- [17759](https://github.com/opensearch-project/OpenSearch/pull/17759)
- [17774](https://github.com/opensearch-project/OpenSearch/pull/17774)
- [17791](https://github.com/opensearch-project/OpenSearch/pull/17791)
- [17884](https://github.com/opensearch-project/OpenSearch/pull/17884)
- [17887](https://github.com/opensearch-project/OpenSearch/pull/17887)
- [17894](https://github.com/opensearch-project/OpenSearch/pull/17894)
- [17923](https://github.com/opensearch-project/OpenSearch/pull/17923)
- [17942](https://github.com/opensearch-project/OpenSearch/pull/17942)
- [17944](https://github.com/opensearch-project/OpenSearch/pull/17944)
- [17951](https://github.com/opensearch-project/OpenSearch/pull/17951)
- [17954](https://github.com/opensearch-project/OpenSearch/pull/17954)
- [17977](https://github.com/opensearch-project/OpenSearch/pull/17977)
- [17996](https://github.com/opensearch-project/OpenSearch/pull/17996)
- [18003](https://github.com/opensearch-project/OpenSearch/pull/18003)
- [18013](https://github.com/opensearch-project/OpenSearch/pull/18013)
- [18017](https://github.com/opensearch-project/OpenSearch/pull/18017)
- [18043](https://github.com/opensearch-project/OpenSearch/pull/18043)
- [18050](https://github.com/opensearch-project/OpenSearch/pull/18050)
- [18059](https://github.com/opensearch-project/OpenSearch/pull/18059)
- [18063](https://github.com/opensearch-project/OpenSearch/pull/18063)
- [18066](https://github.com/opensearch-project/OpenSearch/pull/18066)
- [18069](https://github.com/opensearch-project/OpenSearch/pull/18069)
- [18105](https://github.com/opensearch-project/OpenSearch/pull/18105)
- [18107](https://github.com/opensearch-project/OpenSearch/pull/18107)
- [18189](https://github.com/opensearch-project/OpenSearch/pull/18189)
- [18236](https://github.com/opensearch-project/OpenSearch/pull/18236)
- [21225](https://github.com/opensearch-project/OpenSearch/pull/21225)
- [21239](https://github.com/opensearch-project/OpenSearch/pull/21239)
- [13172](https://github.com/opensearch-project/OpenSearch/pull/13172)
- [14912](https://github.com/opensearch-project/OpenSearch/pull/14912)
- [18034](https://github.com/opensearch-project/OpenSearch/pull/18034)
- [18038](https://github.com/opensearch-project/OpenSearch/pull/18038)
- [18094](https://github.com/opensearch-project/OpenSearch/pull/18094)
- [20509](https://github.com/opensearch-project/OpenSearch/pull/20509)
- [21220](https://github.com/opensearch-project/OpenSearch/pull/21220)
- [21328](https://github.com/opensearch-project/OpenSearch/pull/21328)

For more details on the failed tests refer to [OpenSearch Gradle Check Metrics](https://metrics.opensearch.org/_dashboards/app/dashboards#/view/e5e64d40-ed31-11ee-be99-69d1dbc75083) dashboard.

I just saw what looks like a new kind of failure here:

```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.simple.SimpleSearchIT" -Dtests.method="testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}}" -Dtests.seed=F888A6BE313526C9 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=se-SE -Dtests.timezone=Etc/GMT+0 -Druntime.java=21

SimpleSearchIT > testSimpleTerminateAfterTrackTotalHitsUpToSize {p0={"search.concurrent_segment_search.enabled":"true"}} FAILED
    java.lang.AssertionError: expected:<GREATER_THAN_OR_EQUAL_TO> but was:<EQUAL_TO>
        at __randomizedtesting.SeedInfo.seed([F888A6BE313526C9:161A85B12E222BF5]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at org.opensearch.search.simple.SimpleSearchIT.doTestSimpleTerminateAfterTrackTotalHitsUpTo(SimpleSearchIT.java:435)
        at org.opensearch.search.simple.SimpleSearchIT.testSimpleTerminateAfterTrackTotalHitsUpToSize(SimpleSearchIT.java:443)
```

@prudhvigodithi Any chance this is related to your recent change?

--- a/server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java
+++ b/server/src/internalClusterTest/java/org/opensearch/search/simple/SimpleSearchIT.java
@@ -696,11 +696,8 @@ public class SimpleSearchIT extends ParameterizedStaticSettingsOpenSearchIntegTe
         prepareCreate("idx").setMapping("field", "type=float,doc_values=false").get();
         ensureGreen("idx");
 
-        IndexRequestBuilder indexRequestBuilder = client().prepareIndex("idx");
-
-        for (float i = 9000.0F; i < 20000.0F; i++) {
-            indexRequestBuilder.setId(String.valueOf(i)).setSource("{\"field\":" + i + "}", MediaTypeRegistry.JSON).get();
-        }
+        client().prepareIndex("idx").setSource("{\"field\": 10000.0}", MediaTypeRegistry.JSON).get();
+        refresh("idx");
         String queryJson = "{ \"filter\" : { \"terms\" : { \"field\" : [ 10000.0 ] } } }";
         XContentParser parser = createParser(JsonXContent.jsonXContent, queryJson);
         parser.nextToken();

